### PR TITLE
Fix warehouse error names

### PR DESCRIPTION
**Pull request by Ekrekr**
Fixes https://github.com/dataform-co/dataform-co/issues/7007

Presumably the user knows which warehouse they're using

### CODE REVIEW

File: api/dbadapters/redshift.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export class RedshiftDbAdapter implements IDbAdapter {
       if (options.includeQueryInError) {
         throw new Error(`Error encountered while running "${statement}": ${e.message}`);
       }
-      throw new ErrorWithCause(`Error executing Redshift query: ${e.message}`, e);
```
Comment (BenBirt) [inline api/dbadapters/redshift.ts:77]: we shouldn't remove this, we added it for a reason (check blame if you like, but TL; DR: we wanted to clarify whether errors were in the warehouse, or dataform).

instead we should pass a constructor parameter to the DbAdapter that tells it which context it's being used for, and use that.

Comment (Ekrekr) [reply api/dbadapters/redshift.ts:77]: Ah I see, have done this now

File: api/dbadapters/index.ts [hunk 2]
```diff
@@ -50,7 +48,7 @@ export interface IDbAdapter {
 }
 
 export interface IDbAdapterClass<T extends IDbAdapter> {
-  create: (credentials: Credentials) => Promise<T>;
+  create: (credentials: Credentials, warehouseType?: string) => Promise<T>;
```
Comment (BenBirt) [inline api/dbadapters/index.ts:51]:
```suggestion
  create: (credentials: Credentials, warehouseType: string) => Promise<T>;
```

`warehouseType` isn't optional, handling it is

File: api/dbadapters/redshift.ts [hunk 3]
```diff
@@ -46,19 +51,22 @@ export class RedshiftDbAdapter implements IDbAdapter {
...
       };
       const queryExecutor = new PgPoolExecutor(clientConfig);
-      return new RedshiftDbAdapter(queryExecutor);
+      return new RedshiftDbAdapter(queryExecutor, { warehouseType });
     }
   }
 
-  private constructor(private queryExecutor: PgPoolExecutor, private sshTunnel?: SSHTunnelProxy) {}
+  private constructor(
+    private queryExecutor: PgPoolExecutor,
```
Comment (BenBirt) [inline api/dbadapters/redshift.ts:67]: these should both be `readonly`

File: api/dbadapters/redshift.ts [hunk 4]
```diff
@@ -74,7 +82,10 @@ export class RedshiftDbAdapter implements IDbAdapter {
       if (options.includeQueryInError) {
         throw new Error(`Error encountered while running "${statement}": ${e.message}`);
       }
-      throw new ErrorWithCause(`Error executing Redshift query: ${e.message}`, e);
+      throw new ErrorWithCause(
+        `Error executing ${this.options.warehouseType || ""} query: ${e.message}`,
```
Comment (BenBirt) [inline api/dbadapters/redshift.ts:86]: make `warehouseType` non-optional, and remove the `|| ""`

**Pull request merged**